exchanges/stream/websocket.go

```diff
@@ -475,7 +492,11 @@ func (w *Websocket) trafficMonitor() {
 
 	go func() {
 		var trafficTimer = time.NewTimer(w.trafficTimeout)
-		pause := make(chan struct{})
+		var trafficAuthTimer = time.NewTimer(w.trafficTimeout)
```

Comment:
Could you set up a pinghandler for the auth connection on Kraken? If you send ping to Kraken, it will send a pong, this will satisfy reconnection issues. I'm recycling both connections every 30 seconds because my auth websocket receives no messages. There is one for the unauthenticated endpoint.

Reply:
Kraken send heartbeat message each second on both channels (authorized and non authorized)  if no subscription traffic present. So in default configuration, if You do not unsubscribe manually from (krakenWsOwnTrades, krakenWsOpenOrders) subscriptions it's impossible to get recycling for no messages reason. And it very strange that You can't resubscribe on connection lost. Can You please explain how do you test for a connection braking?

Reply:
Anyway, I added the ping handler on auth ws channel, as you asked. I think in will be helpful if user will unsubscribed from all  Auth subscriptions by API, and in that case it will prevent ws reconnection each 30 seconds, when no traffic and no heartbeat come.

Reply:
Thanks for adding the pinghandler! I was not receiving a heartbeat for auth funnily enough, perhaps that occurs only if data has been sent? I'll try and understand that.
I'll try and investigate why I'm not resubscribing with auth use set to `false` today. I use ipconfig /release and ipconfig/renew

Reply:
You can read about heartbeat in Kraken ws API docs, heartbeat  come when you have subscribed on any event and if there is no new data within one second